Fixed .cryptomator bundle extension registration in OS X

MuscleRumble opened the pull request.

Tillerino commented:
Can you explain the changes / what wasn't working and is working now and why?

I wrote the original file (just copied things / trial and error) and would like to learn more about it.

MuscleRumble commented:
Of course!

I have fixed `CFBundleDocumentTypes` by changing `CFBundleTypeName` to `Cryptomator Vault` and added `LSItemContentTypes` with `org.cryptomator.folder`. In summary: `CFBundleTypeName` is not the identifier (UTI), it's just a description. [Source](https://developer.apple.com/library/mac/documentation/DataManagement/Conceptual/DocBasedAppProgrammingGuideForOSX/ApplicationCreationProcess/ApplicationCreationProcess.html#//apple_ref/doc/uid/TP40011179-CH6-997699)

But registering for an UTI is not enough, because it's not a known file type. That's why it has to be additionally defined as an exported UTI declaration. [Source](https://developer.apple.com/library/mac/documentation/FileManagement/Conceptual/understanding_utis/understand_utis_declare/understand_utis_declare.html)

MuscleRumble commented:
But maybe... before this PR is being merged... we should change the UTI to `org.cryptomator.vault`?

MuscleRumble commented:
Oh and what my motivation was to fix this: Cryptomator vaults/bundles had this `org.cryptomator.folder` name, which was weird, because all users could see that. A more user-friendly description like `Cryptomator Vault` is just more appealing. And the UTI declarations were missing completely (`LSItemContentTypes` should be defined and also exported UTI if it's a completely new one), so that's why.
